### PR TITLE
gp/INFRA-674: input vars and destroy feature

### PR DESCRIPTION
**Pull request by gpuliyar**
The PR covers the following
* Adds input vars capability - the need for it to support complex input variables that can be set by different repositories for the terraform to work successfully. Or else the reusable workflow will have too many input variable definition which might not be generic enough and cause more confusion.
* Adds destroy feature - so terraform resources can be destroyed
* Some clean up around how to handle the exit during a failure. The script is made more strict for failures as the reusable workflow has been tested in multiple repositories and has gone through multiple deployment stage as of now.

### CODE REVIEW

File: .github/workflows/terraform-workflow.yml [hunk 1]
```diff
@@ -123,7 +146,18 @@ jobs:
     steps:
       - uses: actions/checkout@v2
 
-      - run: sudo apt-get install jq moreutils;
+      - run: sudo apt-get install jq jo moreutils;
```
Comment (gabrielhesposito) [inline .github/workflows/terraform-workflow.yml:149]: just curious does this work without `-y` ?

**Review by gabrielhesposito — APPROVED** [review]
lgtm

File: .github/workflows/terraform-workflow.yml (hunk 1 above)
Comment (gpuliyar) [reply .github/workflows/terraform-workflow.yml:149]: @gabrielhesposito - yes, it does work.

Comment (gpuliyar) [reply .github/workflows/terraform-workflow.yml:149]: sharing a run that uses similar scripts - https://github.com/envoy/infra-cloudflare/runs/5270759323?check_suite_focus=true

Comment (gpuliyar) [reply .github/workflows/terraform-workflow.yml:149]: Attaching a screenshot as well for future reference

Screen Shot 2022-02-21 at 7 43 35 AM

Comment (gpuliyar) [reply .github/workflows/terraform-workflow.yml:149]: More screenshots to provide evidence that these installed commands actually work and provide the expected results.

Screen Shot 2022-02-21 at 7 45 05 AM

**Pull request merged**